Remotion/Web/UnitTests/Core/Security/UI/WebSecurityAdapterTests/WebPermissionProviderTestHelper.cs

```diff
@@ -19,108 +19,111 @@
+      _stubUser.Setup (_ => _.User).Returns ("user");
+      _mockPrincipalProvider = new Mock<IPrincipalProvider> (MockBehavior.Strict);
+      _mockPrincipalProvider.Setup (_ => _.GetPrincipal()).Returns (_stubUser.Object).Verifiable();
     }
 
     // methods and properties
 
     public void ExpectHasAccess (Enum[] accessTypeEnums, bool returnValue)
     {
       AccessType[] accessTypes = Array.ConvertAll<Enum, AccessType> (accessTypeEnums, AccessType.Get);
-      Expect.Call (
-          _mockObjectSecurityStrategy.HasAccess (
-              Arg.Is (_mockSecurityProvider),
-              Arg.Is (_stubUser),
-              Arg<IReadOnlyList<AccessType>>.List.Equal (accessTypes)))
-          .Return (returnValue);
+      _mockObjectSecurityStrategy.Setup (
+              _ => _.HasAccess (
+                  _mockSecurityProvider.Object,
+                  _stubUser.Object,
+                  accessTypes))
+          .Returns (returnValue)
+          .Verifiable();
     }
 
     public void ExpectHasStatelessAccessForSecurableObject (Enum[] accessTypeEnums, bool returnValue)
     {
       AccessType[] accessTypes = Array.ConvertAll<Enum, AccessType> (accessTypeEnums, AccessType.Get);
-      Expect
-          .Call (
-              _mockFunctionalSecurityStrategy.HasAccess (
-                  Arg.Is (typeof (SecurableObject)),
-                  Arg.Is (_mockSecurityProvider),
-                  Arg.Is (_stubUser),
-                  Arg<IReadOnlyList<AccessType>>.List.Equal (accessTypes)))
-          .Return (returnValue);
+      _mockFunctionalSecurityStrategy.Setup (
+              _ => _.HasAccess (
+                  typeof (SecurableObject),
+                  _mockSecurityProvider.Object,
+                  _stubUser.Object,
+                  accessTypes))
+          .Returns (returnValue)
+          .Verifiable();
     }
 
     public void ExpectHasStatelessAccessForWxeFunction (Type functionType, bool returnValue)
     {
-      Expect.Call (_mockWxeSecurityAdapter.HasStatelessAccess (functionType)).Return (returnValue);
+      _mockWxeSecurityAdapter.Setup (_ => _.HasStatelessAccess (functionType)).Returns (returnValue).Verifiable();
     }
 
-    public void ReplayAll ()
+    public void VerifyAll ()
     {
-      _mocks.ReplayAll ();
+      _mockSecurityProvider.Verify();
+      _mockObjectSecurityStrategy.Verify();
+      _mockFunctionalSecurityStrategy.Verify();
+      _mockWxeSecurityAdapter.Verify();
+      _stubUser.Verify();
+      _mockPrincipalProvider.Verify();
     }
 
-    public void VerifyAll ()
+    public void VerifyNotAllMocksWhereCalled ()
     {
-      _mocks.VerifyAll ();
+      _mockSecurityProvider.Verify (_ => _.IsNull, Times.Never);
+      _mockObjectSecurityStrategy.Verify();
+      _mockFunctionalSecurityStrategy.Verify();
+      _mockWxeSecurityAdapter.Verify();
+      _stubUser.Verify();
+      _mockPrincipalProvider.Verify (_ => _.GetPrincipal(), Times.Never);
     }
```

Comment:
How did this work before where all mocks were verified?

Reply:
Arrggh... it worked because `SetupResult.For` is not verifiable😫